### PR TITLE
Handle read timeout (i.e. no response from server)

### PR DESCRIPTION
**Pull request by fsat**

### CODE REVIEW

File: conductr_cli/validation.py [hunk 1]
```diff
@@ -169,6 +169,22 @@ def handler(*args, **kwargs):
     return handler
 
 
+def handle_read_timeout_error(func):
+    def handler(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except ReadTimeout as err:
+            log = get_logger_for_func(func)
+            log.error('Timed out waiting for response from the server: {}'.format(err.args[0]))
```
Comment (huntc) [inline conductr_cli/validation.py:178]: Do we not want to convey possible reasons why it timed out e.g. roles being unmatched?

**Comment by fsat** [comment]
Manual test

**Timeout during Conduct Load**
- Start 1 node with a particular role
- Load a bundle requiring different role - sample output below

```
Felixs-MBP:conductr-cli felixsatyaputra$ python3 -m conductr_cli.conduct load ~/workspace/typesafe-fsat/logging-test/target/bundle/logging-test-v0-a101449418187d92c789d1adc240b6d62abf3b5fb30439f4e5816c40e75fcf24.zip
Retrieving bundle...
Retrieving file:///Users/felixsatyaputra/workspace/typesafe-fsat/logging-test/target/bundle/logging-test-v0-a101449418187d92c789d1adc240b6d62abf3b5fb30439f4e5816c40e75fcf24.zip
Loading bundle to ConductR...
Error: Timed out waiting for response from the server: HTTPConnectionPool(host='127.0.0.1', port=9005): Read timed out. (read timeout=30)

```

**Timeout during Conduct Run**
- Start 2 nodes with 1 node assigned with the role we're interested in (e.g. elasticsearch)
- Load the bundle needing the role (e.g. elasticsearch bundle)
- Take down the node with the assigned role (so the bundle can't start)
- Execute conduct run - sample output below

```
Felixs-MBP:conductr-cli felixsatyaputra$ python3 -m conductr_cli.conduct info
ID       NAME                    #REP  #STR  #RUN
c73e4b3  conductr-elasticsearch     1     0     0

Felixs-MBP:conductr-cli felixsatyaputra$ python3 -m conductr_cli.conduct run cond
Bundle run request sent.
Bundle c73e4b35cdfb1083fbe0ff6a7ee7d308 waiting to reach expected scale 1
Error: Timed out: Bundle c73e4b35cdfb1083fbe0ff6a7ee7d308 waiting to reach expected scale 1
```

File: conductr_cli/validation.py (hunk 1 above)
Comment (fsat) [inline conductr_cli/validation.py:178]: I suppose we can do this, but I'd like to rename the function to be something specific to `conduct load` if we decide to do this.

Comment (huntc) [inline conductr_cli/validation.py:178]: I think it is important to convey possible solutions when errors occur. I also think that roles will generate a few of these conditions (they have for me).

Comment (fsat) [inline conductr_cli/validation.py:178]: I have renamed the method to convey that it's being used for `conduct load` only, and added the possible solution when timeout occurs during `conduct load`.

File: conductr_cli/validation.py [hunk 2]
```diff
@@ -169,6 +169,23 @@ def handler(*args, **kwargs):
...
 
 
+def handle_conduct_load_read_timeout_error(func):
+    def handler(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except ReadTimeout as err:
+            log = get_logger_for_func(func)
+            log.error('Timed out waiting for response from the server: {}'.format(err.args[0]))
+            log.error('Please ensure roles required by loaded bundle is assigned to one or more ConductR nodes')
```
Comment (huntc) [inline conductr_cli/validation.py]: How about:

`One possible issue may be that there are not enough resources or machines with the roles that you bundle requires`

File: conductr_cli/validation.py [hunk 3]
```diff
@@ -169,6 +169,24 @@ def handler(*args, **kwargs):
...
 
+def handle_conduct_load_read_timeout_error(func):
+    def handler(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except ReadTimeout as err:
+            log = get_logger_for_func(func)
+            log.error('Timed out waiting for response from the server: {}'.format(err.args[0]))
+            log.error('One possible issue may be that there are not enough resources or machines with the roles '
+                      'that your bundle requires')
```
Comment (fsat) [inline conductr_cli/validation.py:180]: @huntc - error message updated; I changed `you bundle requires` to `your bundle requires`

**Comment by fsat** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;zCe6M8pJ&#x2F;166-handle-load-and-run-timeouts-better"> Handle load and run timeouts better</a>

**Pull request merged**